sockeye/output_handler.py

```diff
@@ -267,3 +270,32 @@ def handle(self,
                              t_input.tokens,
                              t_output.tokens,
                              self.threshold)
+
+
+class BeamStoringHandler(OutputHandler):
+    """
+    Output handler to store beam histories in JSON format.
+
+    :param stream: Stream to write translations to (e.g. sys.stdout).
+    """
+
+    def __init__(self, stream):
+        self.stream = stream
+
+    def handle(self,
+               t_input: inference.TranslatorInput,
+               t_output: inference.TranslatorOutput,
+               t_walltime: float = 0.):
+        """
+        :param t_input: Translator input.
+        :param t_output: Translator output.
+        :param t_walltime: Total wall-clock time for translation.
+        """
+        # If the sentence was max_len split, we may have more than one history
+        for h in t_output.beam_histories:
```

Comment:
this will fail if beam_histories is None. Please add a None check.

Reply:
I would say this is fine if it fails. If this handler is used and no beam is there it's a bug in the code that should be fixed :)

Reply:
ok

Reply:
An assert would be in place, then.